File: app/src/main/java/com/node/util/ScriptUtils.java
package com.node.util;

import android.content.Context;

import com.node.v8.V8Context;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.Arrays;
import java.util.Optional;

public class ScriptUtils {

    private static String readFileFromRawDirectory(Context context, int resourceId) {
        InputStream iStream = context.getResources().openRawResource(resourceId);
        ByteArrayOutputStream byteStream = null;
        try {
            byte[] buffer = new byte[iStream.available()];
            iStream.read(buffer);
            byteStream = new ByteArrayOutputStream();
            byteStream.write(buffer);
            byteStream.close();
            iStream.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
        assert byteStream != null;
        return byteStream.toString();
    }

    public static void require(Context ctx_, V8Context v8ctx_, int resourceId) {
        v8ctx_.eval(readFileFromRawDirectory(ctx_, resourceId));
    }

    public static void bulkEval(V8Context v8ctx_, String ...scripts) {
        Optional<String> script = Arrays.stream(scripts).reduce((sc, c) -> sc + c + "\n");
        script.ifPresent(v8ctx_::eval);
    }
}

File: app/src/main/java/com/node/util/ScriptUtils.kt
package com.node.util

import android.content.Context
import android.os.Build
import android.support.annotation.RequiresApi
import com.node.v8.V8Context
import java.io.ByteArrayOutputStream
import java.io.IOException
import java.util.*

object ScriptUtils {

    private fun readFileFromRawDirectory(context: Context, resourceId: Int): String {
        val iStream = context.resources.openRawResource(resourceId)
        var byteStream: ByteArrayOutputStream? = null
        try {
            val buffer = ByteArray(iStream.available())
            iStream.read(buffer)
            byteStream = ByteArrayOutputStream()
            byteStream.write(buffer)
            byteStream.close()
            iStream.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }

        assert(byteStream != null)
        return byteStream!!.toString()
    }

    fun require(ctx_: Context, v8ctx_: V8Context, resourceId: Int) {
        v8ctx_.eval(readFileFromRawDirectory(ctx_, resourceId))
    }

    @RequiresApi(api = Build.VERSION_CODES.N)
    fun bulkEval(v8ctx_: V8Context, vararg scripts: String) {
        val script = Arrays.stream(scripts).reduce { sc, c -> sc + c + "\n" }
        script.ifPresent { v8ctx_.eval(it) }
    }
}
